internal/endpoint/k8s.go

```diff
@@ -175,23 +178,25 @@ func parseURL(rawurl string) (*k8sURL, error) {
 		Service:   svc,
 		Namespace: ns,
 		Kind:      strings.ToLower(u.Query().Get("kind")),
+		Logger:    log.Scoped("k8sendpoint", "A kubernetes endpoint that represents a service"),
 	}, nil
 }
 
 // namespace returns the namespace the pod is currently running in
 // this is done because the k8s client we previously used set the namespace
 // when the client was created, the official k8s client does not
 func namespace() string {
+	logger := log.Scoped("namespace", "A kubernetes namespace")
 	const filename = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
 	data, err := os.ReadFile(filename)
 	if err != nil {
-		log15.Warn("endpoint: falling back to kubernetes default namespace", "error", filename+" is empty")
+		logger.Warn("endpoint: falling back to kubernetes default namespace", log.String("error", filename+" is empty"))
```

Comment:
```suggestion
		logger.Warn("falling back to kubernetes default namespace", log.String("error", filename+" is empty"))
```
The scope already mentions the endpoint so I think we can drop it